### DISCUSSION TITLE
Binary Switch: "currentValue" is not a supported property (ZW0322)

### DISCUSSION THREAD

**Discussion by ptorrent** [zwave-js/zwave-js#6669] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

I'm trying to set binary value for a figaro Wll module 

I only get one end point

What did you expect to happen instead?

When i call setValue :

```
node.setValue({
	"commandClassName":"Binary Switch",
	"commandClass":37,
	"property":"currentValue",
	"endpoint":0,
	"prevValue":false,
	"propertyName":"currentValue",
	"value":false
}, ...

{
    "status": 5,
    "message": "Binary Switch: \"currentValue\" is not a supported property (ZW0322)"
}
```

So I change the setValue valueId to : 

```
node.setValue({
	"commandClassName":"Binary Switch",
	"commandClass":37,
	"endpoint":0,
	"property":"targetValue",
	"propertyName":"targetValue"
} ,....

And now I've a valid result..
```

Do I miss something ? Why I don't get the targetValue as endpoint ?

### Software versions

Driver (node-zwave-js):  12.4.3

aeotec-z-stick-gen5

### Device information

Manufacturer: Fibaro
Model name: Wall Plug – type B
Node ID: 2

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment, marked as answer]
currentValue is readonly. To control a device, you have to set targetValue.

**Comment by ptorrent** [comment]
Thanks a lot for your answer !

But why I don't get this endpoint from 

```
node.getDefinedValueIDs()
```
?

**Reply by AlCalzone** [reply]
Maybe I can tell if you provide the logfile the issue template asks for :)
